### ISSUE TITLE
Does anyone use the onModify = "some shell command" syntax of Lsyncd?

### ISSUE THREAD

**Issue by axkibe** [axkibe/lsyncd#493]
Hi, does anyone use the onModify = "some shell command" syntax of Lsyncd?

Also known as "Layer 3" in the manual.

If not, I'm thinking of scrapping it from Lsyncd 3.0
However, I would still leave in the ability to write Lua functions to be called on onModify that spawn some custom action. But the "function writer" with the ^something syntax, I'd cut that.

There is a very old issue about it, that it would have to count the current double-quote level to properly backslash-escape the inserted data. Anyway, it gets rather complicated and I tend do solve the issue by rather removing the whole thing.

PS: Other things to be removed from Lsyncd 3 are: forking (to be done by systemd), pidfiles (not needed without forking), command-line sync options (you have to have a config file), the insist option (on by default), exclude and excludeFrom (replaced by more generic filter and filterFrom), the _extra hack and the OSX /dev/events interface (from this I know from issue reports, it is used, by I don't have to ressources (speak an uptodate Apple device) to support it and honestly the will.

Planned new things are: custom signal handlers, per sync log file settings, starting/stopping syncs during runtime, multiple config files, minimum wait time before retrying rsync.

Kind regards, Axel

**Comment by piyushcs** [comment]
I am using onModify in "lbash.lua" example, for verifying if a large file is finished copying all the data. Will it also break?

**Comment by axkibe** [comment]
Yes, but in my opinion it's not difficult to write a Lua function to execute a shell function instead.

**Comment by piyushcs** [comment]
Do you mean i have to create function like functions specified in Config Layer 1 in manual ?
Then using inlet.getEvent( ) i have to get events, call corresponding functions.

**Comment by axkibe** [comment]
No, like Layer 2. I'll post you a comparison as soon I get to do it.

**Comment by piyushcs** [comment]
Oh there, i can define "onCreate = function(event)", right?

**Comment by axkibe** [comment]
also onModify = function(event)

the only thing I'm planning to remove is the

 onModify = "some string ^with ^this ^params";

thingy.

**Comment by piyushcs** [comment]
ok, I got it, i just have to change onModify = function(event), rather than bash command,  i will do it now only so code won't break in 3.0.
Why there is no problem in onCreate, onDelete ?
